common/browser_context_options.go

```diff
@@ -132,3 +134,48 @@ func (b *BrowserContextOptions) Parse(ctx context.Context, opts goja.Value) erro
 	}
 	return nil
 }
+
+// WaitForEventOptions are the options used by
+// the browserContext.waitForEvent API.
+type WaitForEventOptions struct {
+	Timeout     time.Duration
+	PredicateFn goja.Callable
+}
+
+// NewWaitForEventOptions created a new instance of
+// WaitForEventOptions with a default timeout.
+func NewWaitForEventOptions(defaultTimeout time.Duration) *WaitForEventOptions {
+	return &WaitForEventOptions{
+		Timeout: defaultTimeout,
+	}
+}
+
+// Parse will parse the options or a callable predicate function.
+// It can parse only a callable predicate function or an object
+// which contains a callable predicate function and a timeout.
+func (w *WaitForEventOptions) Parse(ctx context.Context, optsOrPredicate goja.Value) error {
+	var isCallable bool
+	if gojaValueExists(optsOrPredicate) {
+		rt := k6ext.Runtime(ctx)
+
+		w.PredicateFn, isCallable = goja.AssertFunction(optsOrPredicate)
+		if isCallable {
+			return nil
+		}
+
+		opts := optsOrPredicate.ToObject(rt)
+		for _, k := range opts.Keys() {
+			switch k {
+			case "predicate":
+				w.PredicateFn, isCallable = goja.AssertFunction(opts.Get(k))
+				if !isCallable {
+					return errors.New("predicate function is not callable")
+				}
+			case "timeout": //nolint:goconst
```

Comment:
I remember a dozen of bugs because of using a wrong string value. So what the linter suggests here makes sense :)

Reply:
Since there's 14 places for this change. I might do them in a new PR, WDYT?

Reply:
https://github.com/grafana/xk6-browser/pull/1044

Reply:
Nice! Thanks.